src/test/java/com/exasol/adapter/dialects/hive/HiveSqlDialectIT.java

```diff
@@ -1,385 +1,652 @@
+        uploadFilesToBucket(driverName, driverPath);
+        final Connection exasolConnection = exasolContainer.createConnectionForUser(exasolContainer.getUsername(),
+                exasolContainer.getPassword());
+        statementExasol = exasolConnection.createStatement();
+        try (final Connection connection = getHiveConnection()) {
+            final Statement statementHive = connection.createStatement();
+            createTableHiveSimple(statementHive);
+            createTableDecimalCast(statementHive);
+            createTableAllDataTypes(statementHive);
+            integrationTestSetupManager.createTestTablesForJoinTests(statementHive, SCHEMA_HIVE, TABLE_JOIN_1,
+                    TABLE_JOIN_2);
         }
+        integrationTestSetupManager.createTestSchema(statementExasol, SCHEMA_EXASOL);
+        createJdbcConnection();
+        integrationTestSetupManager.createAdapterScript(statementExasol, SCHEMA_EXASOL + "." + ADAPTER_SCRIPT_EXASOL,
+                VIRTUAL_SCHEMAS_JAR_NAME_AND_VERSION,
+                Optional.of("%jar /buckets/bfsdefault/default/drivers/jdbc/" + driverName + ";\n"));
+        createVirtualSchema(VIRTUAL_SCHEMA_HIVE_JDBC, Optional.empty());
+        createVirtualSchema(VIRTUAL_SCHEMA_HIVE_JDBC_NUMBER_TO_DECIMAL,
+                Optional.of("hive_cast_number_to_decimal_with_precision_and_scale='36,2'"));
     }
 
-    @Test
-    public void testSetup() throws SQLException {
-        final String query = "SELECT X FROM " + VIRTUAL_SCHEMA + ".T";
-        final ResultSet result = executeQuery(query);
-        assertNextRow(result, Long.valueOf("99"));
+    private static Connection getHiveConnection() throws ClassNotFoundException, SQLException, InterruptedException {
+        Class.forName("org.apache.hive.jdbc.HiveDriver");
+        return DriverManager.getConnection("jdbc:hive2://localhost:" + HIVE_EXPOSED_PORT + "/" + SCHEMA_HIVE,
+                HIVE_USERNAME, HIVE_PASSWORD);
     }
 
-    // Join Tests -------------------------------------------------------------
-    @Test
-    void testInnerJoin() throws SQLException {
-        final String query = "SELECT * FROM  " + VIRTUAL_SCHEMA + ".t1 a INNER JOIN  " + VIRTUAL_SCHEMA
-                + ".t2 b ON a.x=b.x";
-        final ResultSet result = executeQuery(query);
-        assertAll(() -> assertNextRow(result, 2L, "bbb", 2L, "bbb"), () -> assertFalse(result.next()));
+    private static String getPropertyFromFile(final String filePath, final String propertyName) {
+        try (final InputStream inputStream = new FileInputStream(filePath)) {
+            final Properties properties = new Properties();
+            properties.load(inputStream);
+            return properties.getProperty(propertyName);
+        } catch (final IOException exception) {
+            throw new IllegalArgumentException("Cannot access the oracle.properties file or read from it.");
```

Comment:
Turning an `IOException` into an `IllegalArgumentException` makes no sense. Simply throw the IOException. It's test code after all.

Reply:
done